omod-common/src/main/java/org/openmrs/module/webservices/helper/TaskServiceWrapper.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.webservices.helper;
+
+import org.openmrs.scheduler.TaskDefinition;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.api.context.Context;
+
+import java.util.Collection;
+
+/**
+ * Wrapping the required task information between related resource and SchedulerService
+ */
+public class TaskServiceWrapper {
+	
+	public TaskDefinition getTaskById(Integer id) {
+		return Context.getSchedulerService().getTask(id);
+	}
+	
+	public TaskDefinition getTaskByName(String taskName) {
+		return Context.getSchedulerService().getTaskByName(taskName);
+	}
+	
+	public Collection<TaskDefinition> getScheduledTasks() {
+		return Context.getSchedulerService().getScheduledTasks();
+	}
+	
+	public Collection<TaskDefinition> getRegisteredTasks() {
+		return Context.getSchedulerService().getRegisteredTasks();
+	}
+	
+	/**
+	 * Save the new task in the service
+	 * 
+	 * @param task will contain the taskDefinition to be saved
+	 */
+	public void saveTaskDefinition(TaskDefinition task) {
+		Context.getSchedulerService().saveTaskDefinition(task);
+	}
+	
+	/**
+	 * It will delete the task from the service
+	 * 
+	 * @param task will contain the taskDefinition to be deleted
+	 * @throws SchedulerException - It will throw in case of any SchedulerService exceptions
+	 */
+	public void deleteTask(TaskDefinition task) throws SchedulerException {
+		if (Context.getSchedulerService().getTaskByName(task.getName()).getStarted()) {
```

Comment:
What is the use of this logic? Isn't it taken care of by the service layer? Why don't you just Context.getSchedulerService().deleteTask?

Reply:
Nope, SchedulerService is allowed to delete only the shutdown tasks. Otherwise, It threw the error. (I think, Core doesn't care about the task status)
So I just check about the task status before delete that task

Reply:
Even then, that is the business of the service layer. Your duty is to simply call Context.getSchedulerService().deleteTask

Reply:
Okay :blush: